torch/csrc/jit/autodiff.cpp

```diff
@@ -662,6 +689,68 @@ static void liftConstants(Gradient& grad_desc, ReverseDetails& rev_info) {
   }
 }
 
+static void deduplicateSizeCaptures(Gradient& grad_desc, ReverseDetails& rev_info) {
+  Block * primal_block = grad_desc.f->block();
+  const auto usedOnlyInReverse = [primal_block](Value * v) {
+    const auto & uses = v->uses();
+    return std::all_of(uses.begin(), uses.end(),
+                       [primal_block](const Use& u) { return u.user->owningBlock() != primal_block; });
+  };
+  auto captures = getReverseCaptures(grad_desc);
+  value_set capture_set (captures.begin(), captures.end());
+  for (Value * capture : captures) {
+    Node * node = capture->node();
+    if (!node->matches("aten::size(Tensor self) -> int[]")) {
+      continue;
+    }
+    if (usedOnlyInReverse(capture) && capture_set.count(node->input())) {
+      WithInsertPoint insert_guard { *rev_info.reverse_block->nodes().begin() };
```

Comment:
just checking: this inserts after the last node in reverse_block, right?

Reply:
This inserts at the beginning of the `reverse_block` (insert point is the node that will be *after* anything that's inserted).

Reply:
Got it, so it's an `node->insertBefore(insert_guard)` (I should really remember that)